### PR TITLE
UI: Translate Multitrack Video error dialog buttons

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

UI: Translate Multitrack Video error dialog buttons

The Yes and No standard buttons are not translated unless we manually set the translated text ourselves.

### Motivation and Context

A user reported to us in OBS Studio 30.2.0-beta1 that these buttons were not translated. We already use this pattern elsewhere in the codebase.
https://github.com/obsproject/obs-studio/blob/0680b642e9cb85101971e8e03bbcd9b574cff6f7/UI/ui-validation.cpp#L47-L49

### How Has This Been Tested?

I've built this locally on Windows 11, but I could not seem to get this specific dialog to pop up.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
For further cleanup, we could probably move `QMessageBox mb` into the first `if` branch, and change the second branch to use `OBSMessageBox::critical` instead. We don't have an OBSMessageBox function that uses the Warning icon that uses Yes/No buttons, or I would suggest we use that. We could add one, but that was more work than I wanted to do this moment.

File: UI/multitrack-video-error.cpp [hunk 1]
```diff
@@ -31,9 +32,14 @@ bool MultitrackVideoError::ShowDialog(
...
-				      QMessageBox::StandardButton::No);
-		return mb.exec() == QMessageBox::StandardButton::Yes;
+		QAbstractButton *yesButton =
+			mb.addButton(QTStr("Yes"), QMessageBox::YesRole);
+		mb.addButton(QTStr("No"), QMessageBox::NoRole);
+
+		if (mb.clickedButton() != yesButton)
+			return false;
+		else
+			return true;
```
Comment (derrod) [inline UI/multitrack-video-error.cpp:42]: Just use `return mb.clickedButton() == yesButton;`

Comment (RytoEX) [reply UI/multitrack-video-error.cpp]: Done.

**Review by derrod — CHANGES_REQUESTED** [review]
Based on the documentation you'd still need to call `exec`, otherwise thid will always return null: https://doc.qt.io/qt-6/qmessagebox.html#clickedButton

**Comment by RytoEX** [comment]
> Based on the documentation you'd still need to call `exec`, otherwise thid will always return null: https://doc.qt.io/qt-6/qmessagebox.html#clickedButton

Whoops. Fixed.

**Review by derrod — APPROVED** [review]
Seems fine but unable to test right now

**Review by gxalpha — APPROVED** [review]

**Pull request merged**